core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -652,7 +652,9 @@ private boolean isExistingRelation(Session session, QualifiedObjectName name)
             return true;
         }
 
-        return getTableHandle(session, name).isPresent();
+        // If the table is not redirected, check table existence
+        // If the table is redirected, but the target table handle is missing, an exception is thrown
+        return getRedirectionAwareTableHandle(session, name).getTableHandle().isPresent();
```

Comment:
the `isExistingRelation` is currently used only in `io.trino.metadata.MetadataManager#listTables` where we decided not to confirm table existence with the target catalog. The filtered case could be done consistently.
However, we currently have no SPI for checking relation existence without following redirects. `getTableHandle` would be that, but it _also_ does additional checks ("not a Hive table", "not an iceberg table").

It seems we could benefit from consolidated `resolveRelation` SPI which would determine whether relation is  a table, redirect, view or materialized view (#9400 cc @losipiuk).

For now, add a comment and maybe a TODO link. 

Also, please expand the line

```
// If the table is redirected, but the target table handle is missing, an exception is thrown
```

-- currently it looks like the behavior is desireable, while it not necessarily is. It's just documenting current state of things.